### PR TITLE
Apply all EdgeCrafter spatial feature blocks

### PR DESCRIPTION
**Pull request by yutong-xiang-97**
## What has changed and why?

This fixes two related problems in the exported ECSeg segmentation-mask path.

### Segmentation-head deploy path

During ONNX export, the ECSeg transformer was not switched to a segmentation-specific deploy path because it did not implement the expected `convert_to_deploy()` method. The exported path receives only the final decoder query state, but the segmentation head processed the spatial features with only `self.blocks[0]`. The other three trained spatial blocks were skipped, producing incorrect segmentation behavior in the exported model. The stashed `seg head` change reproduces this behavior in the LightlyTrain ECSeg implementation.

The ECSeg transformer now implements `convert_to_deploy()` and enables a deploy-only deferred-mask path. `forward_deploy()` applies every spatial block before projecting the spatial features and pairing them with the final decoder query state. Training and validation retain the existing materialized-mask path used by the matcher and mask losses.

### Expensive `GatherElements` mask path

The previous ONNX mask-output chain was approximately:

```
einsum
→ torch.stack([one element])
→ Gather[-1]
→ GatherElements(index = Tile → (1, 300, 160, 160), int64, approximately 61 MB)
→ masks
```

In the LightlyTrain TensorRT graph this path costs approximately 1.24 ms—around 6× the corresponding EdgeCrafter path. The likely cause is the 61 MB tiled-index `GatherElements`, together with the stack-of-one `Unsqueeze` / `Concat` / `Gather` chain, landing on a slow FP32 or uncoalesced path instead of being fused or folded.

The deploy path now defers the mask einsum. The segmentation head returns projected spatial features, projected query features, and the mask bias. The postprocessor first gathers the selected compact query embeddings and only then computes the mask einsum. This avoids `GatherElements` over the full `(B, Q, Hm, Wm)` mask tensor while preserving the resulting selected masks.

This `GatherElements` fix is based on the `fix gather element` commit from the `debug-instance-segmentation-benchmark` branch.

## How has it been tested?

- The inference benchmarks.

## Did you update [CHANGELOG.md](../CHANGELOG.md)?

- [ ] Yes
- [x] Not needed (internal change)

## Did you update the documentation?

- [ ] Yes
- [x] Not needed (internal change without effects for user)

### CODE REVIEW

File: src/lightly_train/_task_models/instance_segmentation_components/edgecrafter_head.py [hunk 1]
```diff
@@ -185,3 +189,36 @@ def forward(
...
+            self.image_size[1] // self.downsample_ratio,
+        )
+        spatial_features = F.interpolate(
+            spatial_features,
+            size=target_size,
+            mode="bilinear",
+            align_corners=False,
+        )
+        for block in self.blocks:
+            spatial_features = block(spatial_features)
```
[the remark this answers is not in the document]
Comment (yutong-xiang-97) [reply src/lightly_train/_task_models/instance_segmentation_components/edgecrafter_head.py:219]: Will not happen in any of the existing configs.

**Review by gabrielfruet — APPROVED** [review]
LGTM!
